trieste/models/gpflow/inducing_point_selectors.py

```diff
@@ -0,0 +1,224 @@
+        dataset: Dataset,
+    ) -> TensorType:
+        """
+        Calculate the new inducing points given the existing inducing points.
+
+        If `recalc_every_model_update` is set to False then we only generate new inducing points
+        for the first :meth:`update` call, otherwise we just return the current inducing points.
+
+        :param current_inducing_points: The current inducing points used by the model.
+        :param model: The sparse model.
+        :param dataset: The data from the observer.
+        :return: The new updated inducing points.
+        :raise NotImplementedError: If model has more than one set of inducing variables.
+        """
+        tf.debugging.Assert(current_inducing_points is not None, [])
+
+        if isinstance(current_inducing_points, list):
+            raise NotImplementedError(
+                """
+                InducingPointSelectors only currently support models with a single set
+                of inducing points.
+                """
+            )
+
+        if (
+            not self._initialized
+        ) or self._recalc_every_model_update:  # calculate new inducing points when required
+            self._initialized = True
+            M = tf.shape(current_inducing_points)[0]
+            new_inducing_points = self._recalculate_inducing_points(M, model, dataset)  # [M, D]
+            tf.assert_equal(tf.shape(current_inducing_points), tf.shape(new_inducing_points))
+            return new_inducing_points  # [M, D]
+        else:  # otherwise dont recalculate
+            return current_inducing_points  # [M, D]
+
+    @abstractmethod
+    def _recalculate_inducing_points(
+        self, M: int, model: ProbabilisticModelType, dataset: Dataset
+    ) -> TensorType:
+        """
+        Method for calculating new inducing points given a `model` and `dataset`.
+
+        This method is to be implemented by all subclasses of :class:`InducingPointSelector`.
+
+        :param M: Desired number of inducing points.
+        :param model: The sparse model.
+        :param dataset: The data from the observer.
+        :return: The new updated inducing points.
+        """
+        raise NotImplementedError
+
+
+class UniformInducingPointSelector(InducingPointSelector[SupportsGetInducingVariables]):
+    """
+    An :class:`InducingPointSelector` that chooses points sampled uniformly across the search space.
+    """
+
+    def _recalculate_inducing_points(
+        self, M: int, model: SupportsGetInducingVariables, dataset: Dataset
+    ) -> TensorType:
+        """
+        Sample `M` points. If `search_space` is a :class:`Box` then we use a space-filling Sobol
+        design to ensure high diversity.
+
+        :param M: Desired number of inducing points.
+        :param model: The sparse model .
+        :param dataset: The data from the observer.
+        :return: The new updated inducing points.
+        """
+
+        if isinstance(self._search_space, Box):
```

Comment:
what about mixed spaces? we should still do `sample_sobol` on Box subspaces, right?

Reply:
good spot!